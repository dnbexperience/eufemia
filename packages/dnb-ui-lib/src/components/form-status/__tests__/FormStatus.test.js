/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  toJson,
  axeComponent
} from '../../../core/jest/jestSetup'
import Component from '../FormStatus'

const props = fakeProps(require.resolve('../FormStatus'), {
  optional: true
})
props.status = 'error'
props.icon = 'exclamation'

describe('FormStatus component', () => {
  const Comp = mount(<Component {...props} />)

  it('have to match snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('should validate with ARIA rules', async () => {
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })

  it('has to to have a text value as defined in the prop', () => {
    expect(Comp.find('.dnb-form-status--text').text()).toBe('text')
  })
})
