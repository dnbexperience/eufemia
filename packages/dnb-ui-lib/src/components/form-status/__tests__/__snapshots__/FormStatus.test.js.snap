// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStatus component have to match snapshot 1`] = `
<FormStatus
  animation="animation"
  class="class"
  className="className"
  icon="exclamation"
  icon_size="icon_size"
  render_content={[Function]}
  status="error"
  text="text"
  title="title"
>
  <span
    className="dnb-form-status dnb-form-status--error dnb-form-status--animation className class"
    title="title"
  >
    <IconPrimary
      alt={null}
      area_hidden={false}
      attributes={null}
      class={null}
      className={null}
      color={null}
      height={null}
      icon="exclamation"
      modifier={null}
      size="icon_size"
      width={null}
    />
    <span
      className="dnb-form-status--text"
    >
      text
    </span>
  </span>
</FormStatus>
`;
